src/algebra/squarefree.lean

```diff
@@ -450,6 +458,14 @@ lemma squarefree_iff_prime_sq_not_dvd (n : ℕ) :
 squarefree_iff_irreducible_sq_not_dvd_of_exists_irreducible
   ⟨2, (irreducible_iff_nat_prime _).2 prime_two⟩
 
+lemma squarefree_mul {m n : ℕ} (hmn : m.coprime n) :
+  squarefree m ∧ squarefree n ↔ squarefree (m * n) :=
```

Comment:
Do both directions require hmn?

Reply:
No, though the thought was here to mimic the idea of a "multiplicative function." I can create a `squarefree_mul'` that looks like:

```lean
/- Restatement of one direction of `squarefree_mul` without the coprimality assumption -/
lemma squarefree_mul' {m n : ℕ} : squarefree (m * n) → squarefree m ∧  squarefree n := ...
```

Reply:
I think this would be a good idea, `squarefree.of_mul_left` for `squarefree (m * n) -> squarefree m` (perhaps Eric will have a better name)

Reply:
Added with the suggested names.

Reply:
```suggestion
  squarefree (m * n) ↔ squarefree m ∧ squarefree n :=
```
Since we usually name iffs by their LHS